### PR TITLE
Report acked replication task timestamp

### PR DESCRIPTION
**Pull request by yiminc**
**What changed?**
Report back acked replication task timestamp

**Why?**
So source cluster could know how far the remote cluster is lagging behind on replication queue.

**How did you test it?**
Manual test with debugging print showing the replication timestamp is reported back to source.

**Potential risks**
No

**Is hotfix candidate?**
No

### CODE REVIEW

File: service/history/shard/context_impl.go [hunk 1]
```diff
@@ -102,7 +102,13 @@ type (
 		timerMaxReadLevelMap      map[string]time.Time // cluster -> timerMaxReadLevel
 
 		// exist only in memory
-		remoteClusterCurrentTime map[string]time.Time
+		remoteClusterInfos map[string]*remoteClusterInfo
+	}
+
+	remoteClusterInfo struct {
```
Comment (yux0) [inline service/history/shard/context_impl.go:108]: remoteClusterReplicationInfo?

File: service/history/replicatorQueueProcessor.go [hunk 2]
```diff
@@ -386,6 +386,7 @@ func (p *replicatorQueueProcessorImpl) generateSyncActivityTask(
 						VersionHistory:     versionHistory,
 					},
 				},
+				VisibilityTime: &taskInfo.VisibilityTimestamp,
```
Comment (yux0) [inline service/history/replicatorQueueProcessor.go:389]: I don't see this field is set in task_generator.go. worth to double check if we put the timestamp when creating replication tasks.

Comment (yiminc) [reply service/history/replicatorQueueProcessor.go:389]: They are set here:
https://github.com/temporalio/temporal/blob/1198962e9796599c2aabf53057c68f861e577578/service/history/workflow/task_generator.go#L538
And here:
https://github.com/temporalio/temporal/blob/1198962e9796599c2aabf53057c68f861e577578/service/history/workflow/mutable_state_util.go#L52

Comment (yux0) [reply service/history/replicatorQueueProcessor.go:389]: Missed it. LGTM

**Review by yux0 — APPROVED** [review]

**Pull request merged**